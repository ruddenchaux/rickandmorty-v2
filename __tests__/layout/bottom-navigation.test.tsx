import BottomNavigation from '@/app/layout/bottom-navigation';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

jest.mock('next/navigation', () => ({
  usePathname: () => '/',
}));

jest.mock('../../app/layout/items-menu', () => ({
  itemsMenu: [
    {
      label: 'Characters',
      icon: 'img',
      to: '/',
    },
    {
      label: 'Locations',
      icon: 'img',
      to: '/locations',
    },
    ,
  ],
}));

afterEach(() => {
  jest.clearAllMocks();
});

describe('BottomNavigation', () => {
  it('should be rendered with all links', () => {
    const { container } = render(<BottomNavigation />);

    const characters = screen.getByRole('link', { name: 'Characters' });
    const locations = screen.getByRole('link', { name: 'Locations' });

    expect(characters).toBeInTheDocument();
    expect(characters).toHaveAttribute('href', '/');

    expect(locations).toBeInTheDocument();
    expect(locations).toHaveAttribute('href', '/locations');

    expect(container).toMatchSnapshot();
  });
});
