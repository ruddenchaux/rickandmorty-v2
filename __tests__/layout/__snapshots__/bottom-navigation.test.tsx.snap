// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomNavigation should be rendered with all links 1`] = `
<div>
  <div
    aria-label="Bottom navigation"
    class="MuiBottomNavigation-root css-ksfhj9-MuiBottomNavigation-root"
    role="navigation"
  >
    <a
      class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
      href="/"
      tabindex="0"
    >
      <img />
      <span
        class="MuiBottomNavigationAction-label Mui-selected css-imwso6-MuiBottomNavigationAction-label"
      >
        Characters
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
    <a
      class="MuiButtonBase-root MuiBottomNavigationAction-root css-1bjk3jo-MuiButtonBase-root-MuiBottomNavigationAction-root"
      href="/locations"
      tabindex="0"
    >
      <img />
      <span
        class="MuiBottomNavigationAction-label css-imwso6-MuiBottomNavigationAction-label"
      >
        Locations
      </span>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;
